File: PROGRAM/PYTHON/petshop.py
# Saya Faiz Bayu Erlangga dengan NIM 2311231 mengerjakan Tugas Praktikum 1 
# dalam mata kuliah Desain dan Pemrograman Berorientasi Objek untuk keberkahanNya 
# maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin. 

class petshop:

    # ATRIBUT PRIVATE
    __id = 0
    __nama_produk = ""
    __kategori_produk = ""
    __harga_produk = 0

    # CONSTRUCTOR
    def __init__(self, id=0, nama_produk="", kategori_produk="", harga_produk=0):
        self.__id = id
        self.__nama_produk = nama_produk
        self.__kategori_produk = kategori_produk
        self.__harga_produk = harga_produk

    # GETTER
    def get_id(self):
        return self.__id

    def get_nama_produk(self):
        return self.__nama_produk

    def get_kategori_produk(self):
        return self.__kategori_produk

    def get_harga_produk(self):
        return self.__harga_produk

    # SETTER
    def set_id(self, id):
        self.__id = id

    def set_nama_produk(self, nama_produk):
        self.__nama_produk = nama_produk

    def set_kategori_produk(self, kategori_produk):
        self.__kategori_produk = kategori_produk

    def set_harga_produk(self, harga_produk):
        self.__harga_produk = harga_produk

    # Fungsi untuk tambahkan produk
    def tambah(self, id, nama_produk, kategori_produk, harga_produk):
        self.set_id(id)
        self.set_nama_produk(nama_produk)
        self.set_kategori_produk(kategori_produk)
        self.set_harga_produk(harga_produk)

    # Fungsi untuk tampilkan produk
    def tampilkan(self):
        print(f"ID: {self.__id}")
        print(f"Nama: {self.__nama_produk}")
        print(f"Kategori: {self.__kategori_produk}") 
        print(f"Harga: {self.__harga_produk}")

    # FUNGSI EDIT
    def edit(daftar_produk, id, nama_produk, kategori_produk, harga_produk):
        ketemu = 0 # menandai apakah produk ditemukan
        for produk in daftar_produk:
            if produk.get_id() == id: # jika id sama
                produk.set_nama_produk(nama_produk) # set nama baru
                produk.set_kategori_produk(kategori_produk) # set kategori baru
                produk.set_harga_produk(harga_produk) # set harga baru
                ketemu = 1 # tandai jika sudah ditemukan
                break
        if ketemu == 0:
            print("Produk dengan ID ", id, " tidak ditemukan!")


    # FUNGSI HAPUS
    def hapus(daftar_produk, id):
        ketemu = 0 # menandai apakah produk ditemukan
        for produk in daftar_produk:
            if produk.get_id() == id: # jika id sama
                daftar_produk.remove(produk) # menghapus produk menggunakan remove dari library
                print("Produk berhasil dihapus!")
                ketemu = 1 # tandai jika sudah ditemukan
                return
        
        if ketemu == 0:
            print("Produk dengan ID ", id, " tidak ditemukan!")


    # FUNGSI CARI
    def cari(daftar_produk, id):
        for produk in daftar_produk:
            if produk.get_id() == id: # jika id sama
                print("\nProduk ditemukan:")
                produk.tampilkan() # panggil fungsi untuk tampilkan
                return
        print("Produk dengan ID", id, "tidak ditemukan!")

File: petshop.php
<?php
    // ATRIBUT PRIVATE
    class petshop{
        private $id = 0;
        private $nama_produk = '';
        private $kategori_produk = '';
        private $harga_produk = 0;
        private $foto = '';

    // CONSTRUCTOR
    public function __construct($id= 0, $nama_produk = '', $kategori_produk = '', $harga_produk=0, $foto = ''){
        $this->id = $id;
        $this->nama_produk = $nama_produk;
        $this->kategori_produk = $kategori_produk;
        $this->harga_produk = $harga_produk;
        $this->foto = $foto;
    }

    // GETTER
    public function get_id(){
        return $this->id;
    }
    public function get_nama_produk(){
        return $this->nama_produk;
    }
    public function get_kategori_produk(){
        return $this->kategori_produk;
    }
    public function get_harga_produk(){
        return $this->harga_produk;
    }
    public function get_foto() {
        return $this->foto;
    }
    
    // SETTER
    public function set_id($id){
        $this->id = $id;
    }
    public function set_nama_produk($nama_produk){
        $this->nama_produk = $nama_produk;
    }
    public function set_kategori_produk($kategori_produk){
        $this->kategori_produk = $kategori_produk;
    }
    public function set_harga_produk($harga_produk){
        $this->harga_produk = $harga_produk;
    }
    public function set_foto($foto) {
        $this->foto = $foto;
    }

    // Fungsi untuk tampilkan produk
    public function tampilkan(){
        echo "ID: " . $this->id . "<br>";
        echo "Nama: " . $this->nama_produk . "<br>";
        echo "Kategori: " . $this->kategori_produk . "<br>";
        echo "Harga: " . $this->harga_produk . "<br>";
        echo "Foto: <img src='" . $this->foto . "' width='100'><br>";
    }
      
    // Fungsi untuk tambahkan produk
    public static function tambah(&$list, $id, $nama, $kategori, $harga, $foto) {
        $produk = new petshop(); // membuat objek 
        $produk->set_id($id); // set nama
        $produk->set_nama_produk($nama); // set nama
        $produk->set_kategori_produk($kategori); // set kategoi
        $produk->set_harga_produk($harga); // set harga
        $produk->set_foto($foto); // upload foto
        $list[] = $produk; // tambahkan ke daftar
        echo "Produk berhasil ditambahkan!<br>";
    }

    // Fungsi untuk edit produk
    public static function edit(&$list, $id, $nama, $kategori, $harga, $foto) {
        foreach ($list as &$p) { // looping list
            if ($p->get_id() == $id) { // jika id ditemukan
                $p->set_nama_produk($nama); // set nama baru
                $p->set_kategori_produk($kategori); // set kategori baru
                $p->set_harga_produk($harga); // set harga baru
                $p->set_foto($foto); // upload foto baru
                echo "Produk berhasil diperbarui!<br>";
                return;
            }
        }
        echo "Produk dengan ID $id tidak ditemukan!<br>";
    }

    // Fungsi untuk hapus produk
    public static function hapus(&$list, $id) {
        foreach ($list as $key => $p) { // looping list 
            if ($p->get_id() == $id) { // jika id ditemukan
                unset($list[$key]); // hapus menggunakan library dari php 
                echo "Produk berhasil dihapus!<br>";
                return;
            }
        }
        echo "Produk dengan ID $id tidak ditemukan!<br>";
    }

    // Fungsi untuk cari produk
    public static function cari($list, $id) {
        foreach ($list as $p) { // looping list 
            if ($p->get_id() == $id) { // jika id ditemukan
                $p->tampilkan(); // panggil fungsi tampilkan
                echo "Produk berhasil ditemukan!<br>";
                return;
            }
        }
        echo "Produk dengan ID $id tidak ditemukan!<br>";
    }
}
?>